app/views/shared/_navbar.html.erb

```diff
@@ -0,0 +1,33 @@
+<nav class="navbar navbar-expand-lg navbar-light bg-light">
+  <div class="container-fluid">
+    <%= link_to 'POST PAGE', root_path, class:'nav-link fs-4 text-dark' %>
+    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarNavDropdown" aria-controls="navbarNavDropdown" aria-expanded="false" aria-label="Toggle navigation">
+      <span class="navbar-toggler-icon"></span>
+    </button>
+    <div class="collapse navbar-collapse" id="navbarNavDropdown">
+      <ul class="navbar-nav ms-auto">
+        <li class="nav-item">
+          <%= link_to 'Home', root_path, class:'nav-link text-dark' %>
+        </li>
+        
+        <li class="nav-item dropdown">
+          <a class="nav-link dropdown-toggle active" href="#" id="navbarDropdownMenuLink" role="button" data-bs-toggle="dropdown" aria-expanded="false">
+            <% if user_signed_in? %>
+                Welcome <%= current_user.email %>
+             <%else%>
+                Not logged in
+            <%end%>  
+          </a>
+          <ul class="dropdown-menu" aria-labelledby="navbarDropdownMenuLink">
+             <% if user_signed_in? %>
+                <li><%= link_to 'Sign Out', destroy_user_session_path, method: :delete, class:'nav-link' %></li>
+             <%else%>
+                <li><%= link_to 'Register', new_user_registration_path, class:'nav-link' %></li>
+                <li><%= link_to 'Sign In', new_user_session_path, class:'nav-link' %></li>
+             <%end%>
```

Comment:
views should be logic-free. you should not have any logic in the views, specifically `if-else` statements.